jbang catalog list and jbang alias fails if  catalog is giving 404

```
jbang catalog list
[jbang] [ERROR] Could not download https://github.com/maxandersen/jbang/blob/HEAD/jbang-catalog.json
[jbang] Run with --verbose for more details
```

```
[jbang] [ERROR] Could not download https://github.com/maxandersen/jbang/blob/HEAD/jbang-catalog.json
dev.jbang.cli.ExitException: Could not download https://github.com/maxandersen/jbang/blob/HEAD/jbang-catalog.json
	at dev.jbang.source.ResourceRef.fetchFromURL(ResourceRef.java:321)
	at dev.jbang.source.ResourceRef.forResource(ResourceRef.java:229)
	at dev.jbang.source.ResourceRef.forResource(ResourceRef.java:151)
	at dev.jbang.catalog.Catalog.getByRef(Catalog.java:190)
	at dev.jbang.catalog.Catalog.merge(Catalog.java:248)
	at dev.jbang.catalog.Catalog.getMerged(Catalog.java:240)
	at dev.jbang.cli.AliasList.doCall(Alias.java:122)
	at dev.jbang.cli.BaseCommand.call(BaseCommand.java:85)
	at dev.jbang.cli.BaseCommand.call(BaseCommand.java:12)
	at picocli.CommandLine.executeUserObject(CommandLine.java:1953)
	at picocli.CommandLine.access$1300(CommandLine.java:145)
	at picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2352)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2346)
	at dev.jbang.cli.Jbang$3.handle(Jbang.java:135)
	at dev.jbang.cli.Jbang$3.handle(Jbang.java:130)
	at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:2179)
	at picocli.CommandLine.execute(CommandLine.java:2078)
	at dev.jbang.Main.main(Main.java:14)
Caused by: java.io.FileNotFoundException: No file to download. Server replied HTTP code: 404
	at dev.jbang.util.Util.downloadFile(Util.java:592)
	at dev.jbang.util.Util.downloadFile(Util.java:512)
	at dev.jbang.util.Util.downloadAndCacheFile(Util.java:664)
	at dev.jbang.source.ResourceRef.fetchFromURL(ResourceRef.java:318)
	... 17 more
```

i would say we should at most WARN - not fail hard here as it means you can't list aliases for even local files.

--- a/src/main/java/dev/jbang/catalog/Catalog.java
+++ b/src/main/java/dev/jbang/catalog/Catalog.java
@@ -248,9 +248,14 @@ public class Catalog {
 		// Merge the aliases and templates of the catalog refs
 		// into the current catalog
 		for (CatalogRef ref : catalog.catalogs.values()) {
-			Catalog cat = getByRef(ref.catalogRef);
-			result.aliases.putAll(cat.aliases);
-			result.templates.putAll(cat.templates);
+			try {
+				Catalog cat = getByRef(ref.catalogRef);
+				result.aliases.putAll(cat.aliases);
+				result.templates.putAll(cat.templates);
+			} catch (Exception ex) {
+				Util.warnMsg(
+						"Unable to read catalog " + ref.catalogRef + " (referenced from " + catalog.catalogRef + ")");
+			}
 		}
 		result.aliases.putAll(catalog.aliases);
 		result.templates.putAll(catalog.templates);

--- a/src/main/java/dev/jbang/cli/Catalog.java
+++ b/src/main/java/dev/jbang/cli/Catalog.java
@@ -99,12 +99,20 @@ class CatalogUpdate extends BaseCatalogCommand {
 															.entrySet()
 															.stream()
 															.forEach(e -> {
+																String ref = e.getValue().catalogRef;
 																err.println(
 																		"Updating catalog '" + e.getKey()
 																				+ "' from "
-																				+ e.getValue().catalogRef + "...");
-																dev.jbang.catalog.Catalog.getByRef(
-																		e.getValue().catalogRef);
+																				+ ref + "...");
+																try {
+																	dev.jbang.catalog.Catalog.getByRef(ref);
+																} catch (Exception ex) {
+																	Util.warnMsg(
+																			"Unable to read catalog " + ref
+																					+ " (referenced from "
+																					+ e.getValue().catalog.catalogRef
+																					+ ")");
+																}
 															});
 		return EXIT_OK;
 	}

